src/sentry/services/hybrid_cloud/auth/__init__.py

```diff
@@ -174,7 +174,7 @@ def applied_to_request(self, request: Any = None) -> Generator[None, None, None]
 
         old_user = getattr(request, "user", None)
         old_auth = getattr(request, "auth", None)
-        request.user = self.user or AnonymousUser()
+        request.user = self.user or get_user(request)
```

Comment:
The docs indicate that `get_user` will search for user context on the request and if none is found, return an AnonymousUser. I think this can be used as a substitute since i was receiving this error when trying to use the model directly:

```  
File "/Users/leander/dev/sentry/src/sentry/options/__init__.py", line 48, in <module>
    default_store.connect_signals()
  File "/Users/leander/dev/sentry/src/sentry/services/hybrid_cloud/__init__.py", line 124, in method
    with auth_context.applied_to_request(), SiloMode.enter_single_process_silo_context(
  File "/Users/leander/.pyenv/versions/3.8.13/lib/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
  File "/Users/leander/dev/sentry/src/sentry/services/hybrid_cloud/auth/__init__.py", line 160, in applied_to_request
    from django.contrib.auth.models import AnonymousUser
  File "/Users/leander/dev/sentry/.venv/lib/python3.8/site-packages/django/contrib/auth/models.py", line 2, in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
  File "/Users/leander/dev/sentry/.venv/lib/python3.8/site-packages/django/contrib/auth/base_user.py", line 47, in <module>
    class AbstractBaseUser(models.Model):
  File "/Users/leander/dev/sentry/.venv/lib/python3.8/site-packages/django/db/models/base.py", line 103, in __new__
    app_config = apps.get_containing_app_config(module)
  File "/Users/leander/dev/sentry/.venv/lib/python3.8/site-packages/django/apps/registry.py", line 252, in get_containing_app_config
    self.check_apps_ready()
  File "/Users/leander/dev/sentry/.venv/lib/python3.8/site-packages/django/apps/registry.py", line 135, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.
```

Reply:
Docs link: https://docs.djangoproject.com/en/4.1/ref/contrib/auth/#django.contrib.auth.get_user

Reply:
So this definitely should not be `get_user`.  Basically, if `self.user` is None, this must be Anonymous user regardless of what `request.user` is (which is what `get_user` does).  `request.user` might in fact have a user object and be returned from `get_user` but the point is that if `self.user` is none, it must be `AnonymousUser` in all cases.

As for the original issue, it may be sufficient to move the import into the function instead.  But deeper issue is, hmm, the connecting of signals is causing all this grief.  I think that's where the real change needs to happen.

Reply:
Ill take a closer look at this, definitely my changes caused this mess.